src/GitVersionCore/GitPreparer.cs

```diff
@@ -150,6 +150,21 @@ public string GetDotGitDirectory()
             return dotGitDirectory;
         }
 
+        public string GetDotGitDirectoryOfMain()
+        {
+            var dotGitDir = GetDotGitDirectoryOfWorktreeOrMain();
```

Comment:
It feels a bit weird to me that something called "get dot git directory **of main**" invokes a method called "get dot git directory **of worktree or main**". It seems like it should be the other way around?

Reply:
We have two types of directories that can contain `.git`: 

a) a regular git repo with a `.git` subdirectory. This might be a normal repo, or the main working tree if worktrees are used.
b) a linked working tree repo with a `.git` file. This file contains the path to the real `.git` directory which resides under `.git/worktrees/<worktreename>` of the main working tree.

In some cases it doesn't matter if we're dealing with case A or B, but in other cases we need to get the `.git` directory of the main working tree.

I tried to name the methods so that this distinction is clear, but maybe you can think of better names?